lib/appsignal/plug.ex

```diff
@@ -78,6 +78,41 @@ if Appsignal.plug? do
       "#{method} #{path}"
     end
 
+    def extract_sample_data(%Plug.Conn{params: params, host: host,
+      method: method, script_name: script_name, request_path: request_path,
+      port: port, query_string: query_string} = conn) do
+
+      %{
+        "params" => Appsignal.Utils.ParamsFilter.filter_values(params),
+        "environment" => %{
+          "host" => host,
+          "method" => method,
+          "script_name" => script_name,
+          "request_path" => request_path,
+          "port" => port,
+          "query_string" => query_string,
+          "request_uri" => url(conn),
+          "peer" => peer(conn)
+        } |> Map.merge(extract_request_headers(conn))
+      }
+    end
+
+    def extract_request_headers(%Plug.Conn{req_headers: req_headers}) do
+      keys = ~w(
+        accept accept-charset accept-encoding accept-language cache-control
+        connection user-agent from referer range
+      )
```

Comment:
I've based this patch on request headers that are expected to be set in a Phoenix request:

```
[{"host", "localhost:4001"},
 {"cookie", "__ar_v4=U35IKTLTJNEP7GWW6OH3N2%3A20161120%3A90%7CKJIDW6DV4NCMJHZA3MCKTP%3A20161120%3A90%7CQVGTR45XBFCQ3FOIW2JQJS%3A20161120%3A90; _session_id=3800dff4ac0321e743d74537d1da7d0d; autopilot_associated=true; visit_id=59688321b96e6f5d174dcc73; timeframe=7D; incident_status=open; accin-name="},
 {"connection", "keep-alive"}, {"upgrade-insecure-requests", "1"},
 {"accept", "text/html,application/xhtml+xml,application/xml;q=0.9,*/*;q=0.8"},
 {"user-agent",
  "Mozilla/5.0 (Macintosh; Intel Mac OS X 10_12_3) AppleWebKit/602.4.8 (KHTML, like Gecko) Version/10.0.3 Safari/602.4.8"},
 {"referer", "http://localhost:4001/"}, {"accept-language", "en-us"},
 {"accept-encoding", "gzip, deflate"}]
```

There are optional headers (like the x-prefixed ones) that I haven't seen in Phoenix requests, so I'd say we don't add them until we need to. Is there anything you're missing specifically?